### PR TITLE
Complete Js3 week2 homework

### PR DESCRIPTION
**Pull request by Merhawi-gebreslasie**

### CODE REVIEW

File: javascript/javascript3/week2/homework/main.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+const fetchData = () => {
+  fetch(movies)
+    .then((res) => res.json())
+    .then((data) => {
+      const badMovies = data
+        .filter((movie) => movie.rating < 3)
+        .map((movie) => {
+          movie.bad = "bad";
+          return movie;
+        });
```
Comment (ghost) [inline javascript/javascript3/week2/homework/main.js]: We only need to filter the movies so we don't need to use `map` here.

File: javascript/javascript3/week2/homework/main.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      console.log("bad movies: ", badMovies);
+      const filteredBadMovies = badMovies.filter((movie) => movie.year > 2000);
+      console.log("bad movies >2000: ", filteredBadMovies);
+    });
+};
```
Comment (ghost) [inline javascript/javascript3/week2/homework/main.js]: Great work 🙌 
Can you try splitting the last `.then` code block into a chain of smaller `.then` calls ?

File: javascript/javascript3/week2/homework/main.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+fetchData();
+
+async function getResolve(resolveAfter) {
+  return await new Promise((resolve) => {
+    setTimeout(() => resolve(), resolveAfter * 1000);
+  });
+}
```
Comment (ghost) [inline javascript/javascript3/week2/homework/main.js]: We just need to return the promise here. No need for async and await :)

File: javascript/javascript3/week2/homework/main.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          `Longitude: ${crd.longitude}` +
+          `More or less ${crd.accuracy} meters.`
+      );
+    };
+
+    const error = (err) => {
+      reject(`ERROR(${err.code}): ${err.message}`);
+    };
+    navigator.geolocation.getCurrentPosition(success, error);
+  });
```
Comment (ghost) [inline javascript/javascript3/week2/homework/main.js:56]: Greatly done 👍

File: javascript/javascript3/week2/homework/main.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      .then((astronod) => astronod.json())
+      .then((astronod) => {
+        if (astronod) {
+          setTimeout(() => resolve(astronod), 3000);
+        } else {
+          setTimeout(() => reject("fetching astronods goes wrong"), 3000);
+        }
+      });
+  });
+}
```
Comment (ghost) [inline javascript/javascript3/week2/homework/main.js]: We were required to wait then fetch the data, but here we're fetching then waiting.

**Review by ghost — COMMENTED** [review]
Nice job 💪 You show a good grasp of await/async and Promises 🙌

File: javascript/javascript3/week2/homework/main.js (hunk 1 above)
Comment (Merhawi-gebreslasie) [reply javascript/javascript3/week2/homework/main.js]: Hi thanks alot for your interesting feedback.I use map for creating bad movies and then filter to get movies>2000.

**Pull request merged**